VisualizationBase/src/nodes/ViewItemNode.cpp

```diff
@@ -62,4 +64,73 @@ ViewItemNode* ViewItemNode::withReference(Model::Node *reference, int purpose)
 	return result;
 }
 
+ViewItemNode* ViewItemNode::withJson(QJsonObject json, const ViewItem *parent)
+{
+	auto result = new ViewItemNode();
+	result->fromJson(json, parent);
+	return result;
+}
+
+QJsonObject ViewItemNode::toJson(const ViewItem* parent) const
+{
+	QJsonObject result;
+	result.insert("purpose", purpose());
+	result.insert("col", parent->positionOfNode(const_cast<ViewItemNode*>(this)).x());
+	result.insert("row", parent->positionOfNode(const_cast<ViewItemNode*>(this)).y());
+	//If it stores a normal, separately persisted node
+	if (reference() && reference()->manager())
+	{
+		result.insert("reference", reference()->manager()->
+					nodeIdMap().id(reference()).toString());
+		result.insert("type", "NODE");
+	}
+	//If the node handles spacing only
+	else if (!reference() && spacingTarget())
+	{
+		result.insert("target", spacingTarget()->manager()->
+								nodeIdMap().id(spacingTarget()).toString());
+		result.insert("parentCol", parent->positionOfNode(spacingParent()).x());
+		result.insert("parentRow", parent->positionOfNode(spacingParent()).y());
+		result.insert("type", "SPACING");
+	}
+	//If it stores an InfoNode, which is not separately persisted
+	else if (auto infoNode = DCast<InfoNode>(reference()))
+	{
+		result.insert("content", infoNode->toJson());
+		result.insert("target", infoNode->target()->manager()->
+					   nodeIdMap().id(infoNode->target()).toString());
+		result.insert("type", "INFO");
+	}
+	return result;
+}
+
+void ViewItemNode::fromJson(QJsonObject json, const ViewItem* parent)
+{
+	if (!json.contains("type"))
+		return;
+	setPurpose(json["purpose"].toInt());
+	//TODO@cyril Does this way to get a manager for the IDs always work?
```

Comment:
Well no. It will work only if the first loaded manager is the one that is responsible for the nodes that this view talks about. Since you don't really know which manager is the correct one, the right thing to do is to iterate over all of them until you see a manager that can give you a non-null node, when you provide it with the proper id.